lib/post/utils/comment_action_helpers.dart

```diff
@@ -210,3 +248,22 @@ List<ExtendedCommentCardActions> _updateDefaultCommentActionItems(BuildContext c
   }
   return updatedList;
 }
+
+void showReportCommentActionBottomSheet(
+  BuildContext context, {
+  required int commentId,
+}) {
+  showModalBottomSheet(
+    context: context,
+    showDragHandle: true,
+    isScrollControlled: true,
+    builder: (_) => BlocProvider.value(
+      value: context.read<PostBloc>(),
+      child: StatefulBuilder(
```

Comment:
Any reason for using a `StatefulBuilder`?  It doesn't look like you're using `setState`.

Reply:
I do have `setState` in the [dialog](https://github.com/thunder-app/thunder/blob/fb6cfc165b89c013717e76952b5ede3bd4af384b/lib/post/widgets/report_comment_dialog.dart#L105).

Reply:
The `ReportCommentDialog` is a `StatefulWidget`, so it has its own notion of `setState`. I'm not sure if you need the `StatefulBuilder` here. Could be wrong though. 😊

Reply:
I'm not too sure here either but I _think_ what @micahmo mentioned is correct since the child itself is a stateful widget?

Reply:
@micahmo  You're correct. Thanks .